packages/react/src/use-injectable.spec.tsx

```diff
@@ -101,4 +101,20 @@ describe('useInjectable', () => {
     )
     expect(cb).toHaveBeenLastCalledWith('bar')
   })
+  it('supports optional tokens', () => {
+    const foo = token('foo')<string | undefined>()
+    const cb = jest.fn()
+    const Component = () => {
+      cb(useInjectable(foo))
```

Comment:
This line throws due to missing dependency in `DependenciesProvider`

Reply:
I think that instead of:
```typescript
const foo = token('foo')<string | undefined>()
```
we should call it like that:
```typescript
const foo = token('foo')<string>('defaultValue')
```
And then we can get this default value when call useInjectable

Reply:
Conceptually, a token can't have a default value because it's a pure "input", a slot where you have to pass a value. On the other hand, if you need default values - you need a "computation", an injectable: `const foo = injectable('foo', (): string => 'defaultValue')`

Reply:
to support it we can do token like so:
```typescript
export function token<Name extends PropertyKey>(name: Name) {
  return <Type = never>(defaultValue?: Type): InjectableWithName<
    {
      readonly name: Name
      readonly type: Type
      readonly optional: undefined extends Type ? true : false
      readonly children: readonly [
        {
          readonly name: typeof TOKEN_ACCESSOR_KEY
          readonly type: TokenAccessor
          readonly optional: true
          readonly children: readonly []
        }
      ]
    },
    Type
  > => {
    const f = (
      dependencies: {
        readonly [TOKEN_ACCESSOR_KEY]?: TokenAccessor
      } & Record<Name, Type>
    ): Type => {
      const accessor = dependencies[TOKEN_ACCESSOR_KEY]
      if (accessor === undefined) {
        return dependencies[name]
      }
      try {
          return accessor(dependencies, name)
      } catch (error) {
        if (defaultValue === undefined) {
          throw error
        }
        return defaultValue
      }
    }
    f.key = name

    // eslint-disable-next-line no-restricted-syntax
    return f as never
  }
}
```

Reply:
I know we can, the issue is that it goes against the concept of a token.

Reply:
We can make it as special optionalToken.
I find it interesting only in case we have an ability to set default value as i have mentioned here:
https://github.com/raveclassic/injectable-ts/issues/19